tests/unit/models/builder-test.js

```diff
@@ -800,22 +382,61 @@ describe('models/builder.js', function() {
       });
     });
 
-    it('hooks are called in the right order without visualization', function() {
-      return builder.build().then(function() {
-        expect(hooksCalled).to.deep.equal(['preBuild', 'build', 'postBuild', 'outputReady']);
-      });
-    });
 
-    if (experiments.BUILD_INSTRUMENTATION) {
-      it('hooks are called in the right order with visualization', function() {
+    describe('instrumentation hooks', function() {
+      beforeEach(function() {
         process.env.EMBER_CLI_INSTRUMENTATION = '1';
-        buildResults = mockBuildResultsWithHeimdallSubgraph;
+      });
+
+      if (experiments.INSTRUMENTATION) {
+        it('invokes the instrumentation hook if it is preset', function() {
+          addon[experiments.INSTRUMENTATION] = function (instrumentation) {
+            hooksCalled.push('instrumentation');
+            instrumentationArg = instrumentation;
+          };
 
-        return builder.build(null, {}).then(function() {
-          expect(hooksCalled).to.deep.equal(['preBuild', 'build', 'postBuild', 'buildInstrumentation', 'outputReady']);
+          return builder.build(null, {}).then(function() {
+            expect(hooksCalled).to.deep.equal(['preBuild', 'build', 'postBuild', 'instrumentation', 'outputReady']);
```

Comment:
I know I added this test originally, but I think we should probably change it so that `instrumentation` fires last (otherwise our build instrumentation doesn't include `outputReady` invocation times).